src/main/java/seedu/address/model/AppointmentBook.java

```diff
@@ -103,6 +108,16 @@ public void sortAppointments(Comparator<Appointment> comparator) {
         appointments.sortAppointments(comparator);
     }
 
+    /**
+     * Undoes the previous add, delete or edit commands for appointments and returns a copy of the
+     * previous appointment book.
+     */
+    public AppointmentBook undo() {
+        AppointmentBook previousAppointmentBook = new AppointmentBook();
+        previousAppointmentBook.setAppointments(previousAppointmentLists.pop());
+        return previousAppointmentBook;
+    }
```

Comment:
Correct me if I am wrong but do you have to set the `previousAppointmentLists` of `previousAppointmentBook` as well? So that the user can call undo two times in a row?

Reply:
Actually, in `AppointmentBook::resetData`, only the `appointmentList` gets replaced

Reply:
Actually, the main question is just whether the user can call undo two times in a row now?

Reply:
can

Reply:
Ohh,that's good! Then we can merge this.